File: go/linkedlistqueue/linked_list_queue_test.go
package linkedlistqueue

import (
	"testing"
	"../queue"
)

func TestLinkedListQueue(t *testing.T) {
	var q queue.Queue
	q = New()

	size := q.GetSize()
	if size != 0 {
		t.Errorf("Wrong size, expected 0, but got %d", size)
	}

	q.Enqueue("H")
	q.Enqueue("e")
	q.Enqueue("l")
	q.Enqueue("l")
	q.Enqueue("o")

	size = q.GetSize()
	if size != 5 {
		t.Errorf("Wrong size, expected 5, but got %d", size)
	}

	head, err := q.GetFront()
	if err != nil {
		t.Error(err)
	}

	if head != "H" {
		t.Errorf("Wrong head, expected H, bug got %v", head)
	}

	for i := 0; i < 5; i++ {
		q.Dequeue()
	}

	empty := q.IsEmpty()
	if !empty {
		t.Errorf("Should be empty, but no")
	}
}

File: js2/__test__/linkedListQueue.test.ts
import LinkedListQueue from "../linkedListQueue";

describe('LinkedList Queue', () => {
    const queue = new LinkedListQueue<number>();

    test('Base Fn', () => {
        expect(queue.getSize()).toBe(0);
        expect(queue.isEmpty()).toBeTruthy();
    });

    test('Queue operation', () => {
        queue.enqueue(1);
        queue.enqueue(2);
        queue.enqueue(3);
        expect(queue.getSize()).toBe(3);
        expect(queue.getFront()).toBe(1);
        expect(queue.dequeue()).toBe(1);
        expect(queue.getFront()).toBe(2);
        expect(queue.isEmpty()).toBeFalsy();
    });

    test('Throw error', () => {
        queue.dequeue();
        queue.dequeue();
        expect(() => queue.dequeue()).toThrow(Error);
        expect(queue.getSize()).toBe(0);
        expect(() => queue.getFront()).toThrow(Error);
        queue.enqueue(100);
        expect(queue.getFront()).toBe(100);
    });

    test('Console', () => {
        queue.enqueue(200);
        queue.enqueue(300);
        expect(queue.toString()).toBe('Queue: front 100->200->300->Null tail');
    });
});